### ISSUE TITLE
additional functionality.

### ISSUE THREAD

**Issue by atherdon** [ChickenKyiv/loopback-tutor-intern-8#20]
What do you think about creating this functionality:

- [ ] Delete account functionality. user go to their profile and see button with warning - delete your account.
- [ ] Invite other users to website - form with email field. after submit we send a email notification and tell them: 'Hey, oyur friend %FirstName% %lastName% invited you to check our website... + link'
- [ ] Change password
- [ ] Export user related data from database.

**Comment by chauhannishith** [comment]
in 1st one do you want to remove only the user data that is stored or all the data also that is associated with that user. According to me, we can only remove the data which is not related to any payments made in the past as a business cannot delete the transactions
In 2nd one, do you want to keep a track of how many people a user has invited? If no then we only need to send an email which can be done easily.
The 3rd one is simple and doesn't require much to be done as the user can only do that when he is already logged in
For the 4th one I am not sure whether it is required in any way or not. Can you tell me how it can be useful?

**Comment by chauhannishith** [comment]
- [x] Change password

**Comment by chauhannishith** [comment]
I don't understand how to do the last one. Can you explain me how to do it?

**Comment by atherdon** [comment]
> in 1st one do you want to remove only the user data that is stored or all the data also that is associated with that user. According to me, we can only remove the data which is not related to any payments made in the past as a business cannot delete the transactions
yes, we wouldn't delete financial data. only data that related to user. so as test case - when i delete my user - all things that connected to user_id should be dropped. and i'll be able to re-sign in and have empty account.

---
> In 2nd one, do you want to keep a track of how many people a user has invited? If no then we only need to send an email which can be done easily.
only send email. but if you want to track invites - it can be a good feature for affiliate marketing :)

---
> The 3rd one is simple and doesn't require much to be done as the user can only do that when he is already logged in
yeah, it's just a form inside account/profile page

---
> Export
when user click that button - he get a csv file with all data related to him. without ids for sure, just a plan array. if you can specify here what data is related to user - we can discuss in such form we need to export it

**Comment by atherdon** [comment]
i think i also need to check how this works too #29
